src/main/java/Transcendent.java

```diff
@@ -0,0 +1,178 @@
+                        if (indexToBeMarked >= List.listCount) {
+                            Printer.printInvalidCommand();
+                        } else {
+                            List.mark(indexToBeMarked);
+                        }
+                    } catch (NumberFormatException e) {
+                        Printer.printInvalidCommand();
+                    }
+                } else if (command.startsWith("unmark ")) {
+                    String numberString = command.split(" ")[1];
+                    int indexToBeUnmarked;
+                    try {
+                        indexToBeUnmarked = Integer.parseInt(numberString) - 1;
+                        if (indexToBeUnmarked >= List.listCount) {
+                            Printer.printInvalidCommand();
+                        } else {
+                            List.unmark(indexToBeUnmarked);
+                        }
+                    } catch (NumberFormatException e) {
+                        Printer.printInvalidCommand();
+                    }
+                } else {
+                    List.add(command);
+                }
+            }
+        }
+
+    }
+
+    private static class List {
+
+        private static class Task {
+            protected String description;
+            protected boolean isDone;
+
+            public Task(String description) {
+                this.description = description;
+                this.isDone = false;
+            }
+
+            public String getStatusIcon() {
+                return (isDone ? "[X]" : "[ ]");
+            }
+        }
+
+        private static int listCount = 0;
+
+        private static Task[] tasks;
+
+        private static void init() {
+            tasks = new Task[100];
+        }
+
+        private static void add(String taskDesc) {
+            Task newTask = new Task(taskDesc);
+            tasks[listCount] = newTask;
+            listCount += 1;
+            Printer.printAddConfirm(taskDesc);
+        }
+
+        private static void list() {
+            Printer.printSeparator();
+            for (int i = 0; i < listCount; i += 1) {
+                System.out.println((i + 1) + "." + tasks[i].getStatusIcon() + " " + tasks[i].description);
+            }
+            Printer.printSeparator();
+        }
+
+        private static void mark(int toBeMarked){
+            if (tasks[toBeMarked].isDone) {
+                Printer.printAlreadyMarked();
+            } else {
+                tasks[toBeMarked].isDone = true;
+                Printer.printMarkConfirm(toBeMarked);
+            }
+        }
+
+        private static void unmark(int toBeUnmarked) {
+            if (!tasks[toBeUnmarked].isDone) {
+                Printer.printAlreadyUnmarked();
+            } else {
+                tasks[toBeUnmarked].isDone = false;
+                Printer.printUnmarkConfirm(toBeUnmarked);
+            }
+        }
+
+    }
+
+}
```

Comment:
overall, good job, maybe the classes could be separated into different files so it is more readable but well done overall